pilot/pkg/bootstrap/trustbundle.go

```diff
@@ -0,0 +1,176 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package bootstrap
+
+import (
+	"fmt"
+	"reflect"
+	"sync"
+	"time"
+
+	meshconfig "istio.io/api/mesh/v1alpha1"
+	"istio.io/istio/security/pkg/pki/util"
+)
+
+// source is all possible sources of MeshConfig
+type Source int
```

Comment:
isn't this already defined in mesh config? do we need to create our own enum?

Reply:
We don't have an explicit enum in the meshConfig. Also, in addition to the certs and spiffe endpoints configured in MeshConfig, we would also also need to incorporate Citadel root-certs as a source. Better to have our own enum